storage/google_drive/drive_client_wrapper.py

```diff
@@ -0,0 +1,172 @@
+        'mimeType': '',
+    }
+    """
+    folder_id = get_path_id(folder_path)
+    return get_children_by_id(folder_id)
+
+
+def get_folder_id(name, parent_id):
+    log.info('Getting id of folder "{}" under parent with id "{}"...'.format(name, parent_id))
+    response = drive_root_folder.list(
+        q="name='{}' and '{}' in parents and mimeType='application/vnd.google-apps.folder'".format(name, parent_id),
+        spaces='drive',
+        fields='files(id)').execute()
+    files = response.get('files', [])
+    if (len(files) == 0):
+        log.error('Folder "{}" not found under parent with id {}.'.format(name, parent_id))
+        exit(1)
+    if (len(files) > 1):
+        log.error('Multiple folders with name "{}" found under parent with id {}.'.format(name, parent_id))
+        exit(1)
+    log.info('Getting id of folder "{}" under parent with id "{}" - done. Folder id is "{}"'.format(name, parent_id, files[0].get('id')))
+    return files[0].get('id')
+
+def get_path_id(path):
+    folders = _split_path(path)
+    folder_id = get_root_id()
+    for folder in folders:
+        folder_id = get_folder_id(folder, folder_id)
+    return folder_id
+
+def _split_path(path):
+    folders = []
+    while path != "":
+        path, folder = os.path.split(path)
+        if folder != "":
+            folders.append(folder)
+    folders.reverse()
+    return folders
+
+def update_file(source_file_path, target_file_id):
+    media = MediaFileUpload(source_file_path,
+                            resumable=True)
+
+    log.info('Updating file with ID "{}" with source file "{}"...'.format(target_file_id, source_file_path))
+    file = drive_root_folder.update(fileId=target_file_id,
+                                    media_body=media,
+                                    fields='name').execute()
+
+    log.info('Updating file with ID "{}" with source file "{}" - done. File name was "{}"'.format(target_file_id, source_file_path, file.get('name')))
+
+def create_file(source_file_path, target_folder_path, target_file_name=None):
+    if target_file_name == None:
+        target_file_name = os.path.basename(source_file_path)
+
+    target_folder_id = get_path_id(target_folder_path)
+    file_metadata = {
+        'name': target_file_name,
+        'parents': [target_folder_id]
+    }
+    media = MediaFileUpload(source_file_path,
+                            resumable=True)
+
+    log.info('Creating file "{}" in folder "{}" with source file "{}"...'.format(target_file_name, target_folder_path, source_file_path))
+    file = drive_root_folder.create(body=file_metadata,
+                                    media_body=media,
+                                    fields='id').execute()
+    log.info('Creating file "{}" in folder "{}" with source file "{}" - done. File id is "{}"'.format(target_file_name, target_folder_path, source_file_path, file.get('id')))
+
+def create_or_update_file(source_file_path, target_folder_path, target_file_name=None):
```

Comment:
please make:

update_or_create

and an optional argument "recursive=False" say whether the folders should be created if they're not there